cmd/authgear/config/config.go

```diff
@@ -0,0 +1,112 @@
+package config
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/json"
+
+	"github.com/lestrrat-go/jwx/jwk"
+
+	"github.com/skygeario/skygear-server/pkg/auth/config"
+	"github.com/skygeario/skygear-server/pkg/core/uuid"
+)
+
+func NewAppConfigFromOptions(opts *Options) *config.AppConfig {
+	return &config.AppConfig{
+		ID:   config.AppID(opts.AppID),
+		HTTP: &config.HTTPConfig{Hosts: opts.Hosts},
+	}
+}
+
+func NewSecretConfigFromOptions(opts *SecretOptions) *config.SecretConfig {
+	var items []config.SecretItem
```

Comment:
Based on the generated secret.yaml, can we update the JSON schema to reflect the actual shape of the various JWK ?

Reply:
Since JWK is an extensible standard, I think we should just add fields that we expect to see, i.e. required fields.